src/core.c/REPL.pm6

```diff
@@ -440,6 +440,12 @@ do {
 
             $!history-file.absolute
         }
+
+        method here() {
+            my $repl := self.new(nqp::getcomp("Raku"),%_);
+            nqp::bindattr($repl,REPL,'$!save_ctx',nqp::ctxcaller(nqp::ctx));
```

Comment:
This isn't going to go well. The compiler is free to optimize out lexicals as it pleases. We do have a list of magical sub names, such as `EVAL`, which inhibit compiler optimization in all outer scopes. (Others include `eval-lives-ok` for `Test`.) I'm not thrilled about this, but at least hope that list can go away in the future by making all such things instead be macros that do...something...to inhibit the optimizations (the point being that the compiler doesn't need to keep a magical list of names).

But that means the things in question should be *sub* calls today, *not* method calls. We do, I grant, have a `.EVAL` method, which I consider an unfortunate mistake, although we could make that a `macro postfix:<.EVAL>` or something (and hope it wins on the LTM longest literal prefix rule). (I'd personally lean towards deprecation, though.)

Anyway, if we really should have this, please make it a `sub`. It'll have to go in the same magical list of optimization inhibitors that `EVAL` is also in.

Reply:
OK, I will make it a sub called `repl`.

@jnthn where does the magical sub names list live?

Reply:
> OK, I will make it a sub called repl.

That's succinct.

> where does the magical sub names list live?

`src/Perl6/Optimizer.nqp` I believe.

Reply:
Aah, that's why I couldn't find it: `eval-lives-ok` is **not** in that list, but `throws-like` is.